sql/moz-fx-data-shared-prod/google_ads_derived/conversion_event_categorization_v1/query.sql

```diff
@@ -0,0 +1,131 @@
+--STEP 1: Get clients with a first seen date = submission date - 14 days
+--Note: Min cohort date is 2023-11-01 so backfilling will return nothing before then
+--Note: Max cohort date cannot be more than 7 days ago (to ensure we always have at least 7 days of data)
+WITH clients_first_seen_14_days_ago AS (
+  SELECT
+    client_id,
+    first_seen_date,
+    country,
+    attribution_campaign,
+    attribution_content,
+    attribution_dltoken,
+    attribution_medium,
+    attribution_source
+  FROM
+    `moz-fx-data-shared-prod.telemetry.clients_first_seen` --contains all new clients, including those that never sent a main ping
+  WHERE
+    first_seen_date = DATE_SUB(@submission_date, INTERVAL 14 DAY)
+    AND first_seen_date
+    BETWEEN '2023-11-01'
+    AND DATE_SUB(CURRENT_DATE, INTERVAL 8 DAY)
+),
+--Step 2: Get only the columns we need from clients last seen, for only the small window of time we need
+clients_last_seen_raw AS (
+  SELECT
+    cls.client_id,
+    cls.first_seen_date,
+    cls.country,
+    cls.submission_date,
+    cls.days_since_seen,
+    cls.active_hours_sum,
+    cls.days_visited_1_uri_bits,
+    cls.days_interacted_bits,
+    cls.search_with_ads_count_all
+  FROM
+    `moz-fx-data-shared-prod.telemetry.clients_last_seen` cls
+  JOIN
+    clients_first_seen_14_days_ago clients
+    ON cls.client_id = clients.client_id
+  WHERE
+    cls.submission_date >= '2023-11-01' --first cohort date
+    AND cls.submission_date
+    BETWEEN cls.first_seen_date
+    AND DATE_ADD(cls.first_seen_date, INTERVAL 6 DAY) --get first 7 days from their first main ping
+    --to process less data, we only check for pings between @submission date - 15 days and submission date + 15 days for each date this runs
+    AND cls.submission_date
+    BETWEEN DATE_SUB(@submission_date, INTERVAL 15 DAY)
+    AND DATE_ADD(@submission_date, INTERVAL 15 DAY)
+),
+--STEP 2: For every client, get the first 7 days worth of main pings sent after their first main ping
+client_activity_first_7_days AS (
+  SELECT
+    client_id,
+    ANY_VALUE(
+      first_seen_date
+    ) AS first_seen_date, --date we got first main ping (potentially different than above first seen date)
+    ANY_VALUE(
+      CASE
+        WHEN first_seen_date = submission_date
+          THEN country
+      END
+    ) AS country, --any country from their first 7 days of main pings
```

Comment:
if CLS is accurate, this should return just one row. but we are aggregating

Reply:
You're right because https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/telemetry_derived/clients_last_seen_v2/query.sql#L117 just keeps repeating whatever country was first - will update the comment - thanks!